src/uct/tcp/tcp_sockcm_ep.c

```diff
@@ -365,6 +472,12 @@ ucs_status_t uct_tcp_sockcm_ep_server_handle_data_received(uct_tcp_sockcm_ep_t *
                                                   UCT_TCP_SOCKCM_EP_DATA_RECEIVED));
 
         ucs_assert(hdr->length == 0);
+
+        if (cep->state & UCT_TCP_SOCKCM_EP_GOT_DISCONNECT) {
```

Comment:
aren't we checking this already in uct_tcp_sockcm_ep_recv() ?

Reply:
right. so actually it also applies to your previous comment - https://github.com/openucx/ucx/pull/5265#discussion_r437322650

Reply:
yes, it should be enough to check once